src/sidebar/components/test/UserMenu-test.js

```diff
@@ -208,6 +211,55 @@ describe('UserMenu', () => {
     });
   });
 
+  describe('open profile item', () => {
+    [{ isFeatureEnabled: true }, { isFeatureEnabled: false }].forEach(
+      ({ isFeatureEnabled }) => {
+        it('includes profile item only for third party users with the feature flag enabled', () => {
+          fakeIsFeatureEnabled.returns(isFeatureEnabled);
+
+          const wrapper = createUserMenu();
+          const openProfileItem = findMenuItem(wrapper, 'Your profile');
+
+          assert.equal(isFeatureEnabled, openProfileItem.exists());
+        });
+      }
+    );
+
+    it('opens the profile when clicked', () => {
+      fakeIsFeatureEnabled.returns(true);
+      fakeIsThirdPartyUser.returns(true);
+
+      const wrapper = createUserMenu();
+      const openProfileItem = findMenuItem(wrapper, 'Your profile');
+
+      openProfileItem.props().onClick();
+      assert.calledOnce(fakeFrameSync.notifyHost);
+      assert.calledWith(fakeFrameSync.notifyHost, 'openProfile');
+    });
+
+    [{ featureIsEnabled: true }, { featureIsEnabled: false }].forEach(
+      ({ featureIsEnabled }) => {
+        it('opens the profile and closes itself when `p` is typed if the feature is enabled', () => {
+          fakeIsFeatureEnabled.returns(featureIsEnabled);
+          fakeIsThirdPartyUser.returns(true);
+
+          const wrapper = createUserMenu();
+          // Make the menu "open"
+          act(() => wrapper.find('Menu').props().onOpenChanged(true));
+          wrapper.update();
+          assert.isTrue(wrapper.find('Menu').props().open);
+
+          wrapper
+            .find('[data-testid="user-menu"]')
+            .simulate('keydown', { key: 'p' });
+
+          assert.equal(featureIsEnabled, fakeFrameSync.notifyHost.called);
+          assert.equal(!featureIsEnabled, wrapper.find('Menu').props().open);
+        });
+      }
+    );
+  });
```

Comment:
This test is hard to human-parse. I'm not sure it benefits from conflating two test cases. Even though it's not DRY, you might separate into two. Or, perhaps update the test description to front-load what's really under test here, as I understand it: "responds to `p` keypress" (or not).